test/fixtures/09-actions/actions.ts

```diff
@@ -0,0 +1,67 @@
+import { actions, errors } from '@balena/pinejs';
+import type ActionsUniversityModel from './actions-university.js';
+import './translations/v1/actions.js';
+
+declare module '../../../out/sbvr-api/sbvr-utils.js' {
+	export interface API {
+		actionsUniversity: PinejsClient<ActionsUniversityModel>;
+	}
+}
+
+actions.addAction(
+	'actionsUniversity',
+	'student',
+	'promoteToNextSemester',
+	async ({ id, api, request }) => {
+		if (id == null) {
+			throw new errors.BadRequestError(
+				'Can only promote one student at a time',
+			);
+		}
+
+		const { grades } = request.values;
+
+		if (!Array.isArray(grades)) {
+			throw new errors.BadRequestError('Invalid payload');
+		}
+
+		const average = grades.reduce((a, b) => a + b) / grades.length;
+
+		if (average < 7) {
+			await api.patch({
+				resource: 'student',
+				id,
+				body: {
+					is_repeating: true,
+					previous_year_grade: `${average}`,
+				},
+			});
+		} else {
+			const student = await api.get({
+				resource: 'student',
+				id,
+				options: {
+					$select: ['semester'],
+				},
+			});
+
+			await api.patch({
+				resource: 'student',
+				id,
+				body: {
+					semester: student!.semester + 1,
```

Comment:
Maybe we can bring assertExists here as well.

Reply:
done